### PR TITLE
Bump k8s go sdk client

### PR DESCRIPTION
**Pull request by liranbg**
Required changes made on `custom_metrics` to receive both `Group` and `GroupKind` in order to use new k8s `custom_metrics`. Thanks to https://github.com/v3io/scaler-types/pull/9/files it is now possible.

- added github actions to lint / test / build
- bump k8s go sdk client to 1.15.5
- bumped scaler-types to 1.7.0 to support `GroupKind` instance (as oppose to groupKind as a string)

### CODE REVIEW

File: cmd/autoscaler/app/autoscaler.go [hunk 1]
```diff
@@ -75,15 +83,18 @@ func createAutoScaler(restConfig *rest.Config,
 		return nil, errors.Wrap(err, "Failed to initialize root logger")
 	}
 
-	customMetricsClient, err := custommetricsv1.NewForConfig(restConfig)
+	dc, err := discovery.NewDiscoveryClientForConfig(restConfig)
```
Comment (omesser) [inline cmd/autoscaler/app/autoscaler.go:86]: please use words

File: cmd/autoscaler/app/autoscaler.go [hunk 2]
```diff
@@ -75,15 +83,18 @@ func createAutoScaler(restConfig *rest.Config,
...
 	if err != nil {
-		return nil, errors.Wrap(err, "Failed create custom metrics client set")
+		return nil, err
```
Comment (omesser) [inline cmd/autoscaler/app/autoscaler.go:88]: please wrap, wrapping preserves the error stack (in @pavius 's nuclio errors)

File: cmd/autoscaler/app/autoscaler.go [hunk 3]
```diff
@@ -75,15 +83,18 @@ func createAutoScaler(restConfig *rest.Config,
...
 	}
+	availableAPIsGetter := custom_metrics.NewAvailableAPIsGetter(dc)
+	mapper := restmapper.NewDeferredDiscoveryRESTMapper(memory.NewMemCacheClient(dc))
+	customMetricsClient := custom_metrics.NewForConfig(restConfig, mapper, availableAPIsGetter)
```
Comment (omesser) [inline cmd/autoscaler/app/autoscaler.go:92]: All this to get `customerMetricsClient`?
I have no idea if there's an easier shorter way, so I'm counting on you there isn't

Comment (liranbg) [reply cmd/autoscaler/app/autoscaler.go:92]: Indeed. After investigation of mine and @Hedingber - this is the way we've found to create the `customMetricsClient` , given the fact we have `restConfig`

**Pull request merged**